components/calendar/src/calendar_arithmetic.rs

```diff
@@ -314,20 +315,14 @@ impl<C: DateFieldsResolver> ArithmeticDate<C> {
                 },
             },
             (Some(era), Some(era_year)) => {
-                let era_year_as_year_info = calendar
-                    .year_info_from_era(era, range_check(era_year, "year", VALID_YEAR_RANGE)?)?;
+                let year = calendar.year_info_from_era(era, era_year)?;
```

Comment:
Issue: The `era_year` shouldn't get plumbed into the calendar impl until we've checked that it is in some reasonable range, maybe +/- 1e7, to allow calendars like simplified chinese to make assumptions that years aren't too big.